README.md

```diff
@@ -251,9 +251,98 @@ Congratulations! You can now send remote push notifications to your device from
+	###### Overriding Whether to Show or Skip a Particular InApp Message
+	When an inApp message arrives from the server, the `onNew` method of `IterableInAppDelegate` is called. This delegate is set via the `inAppDelegate` property of `IterableConfig`. You can set `IterableConfig.inAppDelegate` to a custom class to override the default behavior. This class just needs to implement the `onNew` method. The `onNew` method should return `.show` to show the message or `.skip` to not show the message at this time.
+	
+	Swift:
+
+	```
+	class YourCustomInAppDelegate : IterableInAppDelegate {
+		func onNew(message: IterableInAppMessage) -> InAppShowResponse {
+			// perform custom processing
+
+			// ...
+			
+			return .show // or .skip
+		}
+	}
+	
+	// ...
+	
+	let config = IterableConfig()
+	config.inAppDelegate = YourCustomInAppDelegate()
+	IterableAPI.initialize(apiKey: "YOUR API KEY",  launchOptions: nil, config: config)
+	```
+
+	Objective-C:
+
+	```
+	// Implement this method in your custom class that implements IterableInAppDelegate
+	// This will most likely be the global AppDelegate class.
+	- (enum InAppShowResponse)onNewMessage:(IterableInAppMessage * _Nonnull)message {
+   		// perform custom processing
+   		
+   		// ...
+   		
+    	return InAppShowResponseShow; // or InAppShowResponseSkip
+	}
+
+	// ...
+	
+	// Now set this custom class in IterableConfig
+	IterableConfig *config = [[IterableConfig alloc] init];
+	config.inAppDelegate = self; // or other class implementing the protocol
+	[IterableAPI initializeWithApiKey:@"YOUR API KEY" launchOptions:launchOptions config:config];
+	```
+
+	###### Getting the Local Queue of InApp Messages
+	All inApp messages that arrive from the server are stored in a local queue until they are consumed. IterableSDK exposes `InAppManager` protocol via the `IterableAPI.inAppManager` read-only property to get to locally stored inApp messages. Please be aware that all inApp messages that are shown will be consumed and removed from this queue by default. So you will have to override the default behavior as mentioned above to keep inApp messages around even after they are shown.
+	
+	Swift:
+	
+	```
+	// Get the inApp messages list
+	let messages = IterableAPI.inAppManager.getMessages()
+	
+	// Show an inApp message 
+	IterableAPI.inAppManager.show(message: message)
+	
+	// Show an inApp message without consuming, i.e., not removing it from the queue
+	IterableAPI.inAppManager.show(message: message, consume: false)
+	
+	```	
+	
+	Objective-C:
+	
+	```
+	// Get the inApp messages list
+    NSArray *messages = [IterableAPI.inAppManager getMessages];
+	
+	// Show an inApp message 
+	[IterableAPI.inAppManager showMessage:message];
+	
+	// Show an inApp message without consuming, i.e., not removing it from the queue
+	[IterableAPI.inAppManager showMessage:message consume:NO callbackBlock:nil];
+	
+	```	
+
+	###### When User Clicks a Button in the InApp Message
```

Comment:
- No indent necessary
- Capitalization: When user clicks a button in the in-app message